src/rust/ide/json-rpc/src/handler.rs

```diff
@@ -195,6 +203,11 @@ impl<Notification> {
             channel::emit(event_transmitter,event)
         }
     }
+
+    /// All requests timeout after this duration.
+    pub fn timeout(&self) -> Duration {
+        self.timeout
+    }
```

Comment:
I don't understand: what is "this duration"?

Reply:
It's a type from std. You can see it's docs for more info.

Reply:
Oh, I got `timeout` as a noun, not a verb!
```suggestion
    /// A `Duration` after which requests are timed out.
    pub fn timeout(&self) -> Duration {
        self.timeout
    }
```